xds/internal/xdsclient/authority.go

```diff
@@ -152,8 +173,10 @@ func (a *authority) updateResourceStateAndScheduleCallbacks(rType xdsresource.Ty
 			// Cancel the expiry timer associated with the resource once a
 			// response is received, irrespective of whether the update is a
 			// good one or not.
-			state.wTimer.Stop()
-			state.wState = watchStateRespReceived
+			if state.wState == watchStateRespRequested {
```

Comment:
Is this check required here? If so, what happens when `state.wState != watchStateRespRequested`?

Reply:
I think here is a scenario. The update callback could be called by the transport after a watch for the resource was cancelled.